test/jdk/sun/security/ssl/CipherSuite/TLSCipherSuiteWildCardMatchingIllegalArgument.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8341964
+ * @summary Add mechanism to disable different parts of TLS cipher suite
+ * @run testng/othervm TLSCipherSuiteWildCardMatchingIllegalArgument
+ */
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.fail;
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.security.Security;
+
+import javax.net.ssl.SSLContext;
+
+/**
+ * SSLContext loads "jdk.tls.disabledAlgorithms" system property statically
+ * when it's being loaded into memory, so we can't call
+ * Security.setProperty("jdk.tls.disabledAlgorithms") more than once per test
+ * class. Thus, we need a separate test class each time we need to modify
+ * "jdk.tls.disabledAlgorithms" config value for testing.
+ */
```

Comment:
Still nitpick-level and it might violate coding rules I'm not aware of: Declaring `wildCardMatch` package visible should allow you to test that methods's behaviour by simply calling it with fitting parameters in the other test class. Pass the cache-Map as parameter instead of accessing the "global variable" in the method and you can additionally test the caching mechanism.

Reply:
Yes, I thought about that as well. But I think modifying method's visibility just to accommodate tests goes against current JDK coding conventions. @seanjmullan May correct me on that.

Reply:
Right, changing the method's visibility just to accommodate testing is not a good idea. It can be done in other ways, ex: using reflection but I don't see a compelling enough reason to test the internal methods here.